### ISSUE TITLE
Gamepad.notifyOnConnect listeners are not called on connection or disconnection of gamepads

### ISSUE THREAD

**Issue by dmitry-ant** [Kode/Kha#1289]
**Describe the bug**
After calling Gamepad.notifyOnConnect(), listeners specified are never called, even though Gamepad.get() with hardcoded indices returns working Gamepad instances.

**To Reproduce**
1. Call Gamepad.notifyOnConnect() with valid listeners
2. Connect or disconnect gamepad
3. Listeners are not called

**Expected behavior**
calling Gamepad.notifyOnConnect() causes specified listeners to be called when gamepad is connected or disconnected

**Execution Environment:**
- Host system: Arch Linux x86_64 kernel 5.10.11
- Target system: Arch Linux x86_64 kernel 5.10.11
- IDE used: Visual Studio Code-OSS
- Kha revision: fd61f8ebbd18b6d9971c0407f6877f6a9d010969
- Kha build output: 
```
Using Kha (fd61f8eb, M Backends/Kore/khacpp, M Kinc) from /home/sigma0/lib/Kha
Creating Kha project.
Exporting asset 1 of 5 (animtest.png).
Exporting asset 2 of 5 (checkers.png).
Exporting asset 3 of 5 (p.png).
Exporting asset 4 of 5 (jbmono.ttf).
Exporting asset 5 of 5 (pa.png).
Compiling shader 1 of 8 (painter-colored.frag.glsl).
Compiling shader 2 of 8 (painter-colored.vert.glsl).
Compiling shader 3 of 8 (painter-image.frag.glsl).
Compiling shader 4 of 8 (painter-image.vert.glsl).
Compiling shader 5 of 8 (painter-text.frag.glsl).
Compiling shader 6 of 8 (painter-text.vert.glsl).
Compiling shader 7 of 8 (painter-video.frag.glsl).
Compiling shader 8 of 8 (painter-video.vert.glsl).
Using Kinc (0fe88444) from /home/sigma0/lib/Kha/Kinc
kincfile found.
Creating Linux project files.
Compiling...
Build time: 0m 2s
```
- Application output (if it runs):
```
Starting Kore
Starting application
Application stopped
```
**Additional context**
I would settle for Gamepad.get() with hardcoded indices, but they do not reconnect when gamepad connection is restored after disconnect

**Comment by dmitry-ant** [comment]
is the total silence for a month because of an obscure target? maybe someone could point me to relevant files? is fixing this at all possible? are there workarounds for this?

i am sorry for pestering with this comment, but i really need this functionality.

**Comment by RobDangerous** [comment]
This is a hobby project, maintained by only a single person. One month is nothing.
The code is here: https://github.com/Kode/Kinc/blob/master/Backends/System/Linux/Sources/kinc/backend/input/gamepad.cpp

**Comment by dmitry-ant** [comment]
Well i got something working, but i'm not sure if solution is at all acceptable.

1. It requires linking against udev from systemd client libraries
2. select() call, used for checking read readiness (i guess in case udev fires off faster than libinput or something), blocks until there is something to read, BUT it works without a select() call, not sure if that's stable though
3. i didn't find where to put udev_unref() call, don't know if that is a problem

should i make a PR or is it better to just continue investigating?

**Comment by RobDangerous** [comment]
Sounds more complicated than I expected but please always feel free to send pull-requests even if something isn't necessarily ready to be merged yet - that way I can help figuring things out.

**Comment by dmitry-ant** [comment]
Fixed by [b4e753f](https://github.com/Kode/Kha/commit/b4e753f61a1940c55e852c80d9edad2e3ad1bd9e)
